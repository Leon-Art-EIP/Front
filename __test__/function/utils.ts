import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
