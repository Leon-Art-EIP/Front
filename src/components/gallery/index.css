.leon-title {
    color: #E11C0A;
}

.art-title {
    color: #000000
}

.redirection-button {
    border-radius: 30px;
    box-shadow: #F5F5F5 0px 0px 5px 0px;
    padding-inline: 15px;
    padding-block: 10px;
    border-width: 1px;
    border-color: #b6b6b6;
    color: #E11C0A;
    cursor: pointer;
}

.redirection-button:hover {
    background-color: #fcfcfc;
    transition: transform 0.1s ease-out;
    transform: scale(1.030);
}