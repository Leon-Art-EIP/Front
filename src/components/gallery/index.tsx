import React, { useState } from 'react';
import './index.css';

function Gallery({ redirectUrl, redirectText }: { redirectUrl: string, redirectText: string }) {

  const [pictures, setPictures] = useState<string[]>([]);

  React.useEffect(() => {
    const seedData = [
      "https://irisphoto.art/web/image/65508/19-98-31.jpg",
      "https://tds-images.thedailystar.net/sites/default/files/styles/amp_metadata_content_image_min_696px_wide/public/images/2022/10/14/ai_art_generator.png?itok=kgyM3PUE",
      "https://media.cdnws.com/_i/119489/433/3867/37/jm-arthot-newlessables-044-liberte-time-workofart-frame.jpeg ",
      "https://cdn.pixabay.com/photo/2017/08/30/12/45/girl-2696947_1280.jpg",
      "https://tds-images.thedailystar.net/sites/default/files/styles/amp_metadata_content_image_min_696px_wide/public/images/2022/10/14/ai_art_generator.png?itok=kgyM3PUE",
      "https://cdn.pixabay.com/photo/2017/08/30/12/45/girl-2696947_1280.jpg",
      "https://tds-images.thedailystar.net/sites/default/files/styles/amp_metadata_content_image_min_696px_wide/public/images/2022/10/14/ai_art_generator.png?itok=kgyM3PUE",
      "https://irisphoto.art/web/image/65508/19-98-31.jpg",
      "https://media.cdnws.com/_i/119489/433/3867/37/jm-arthot-newlessables-044-liberte-time-workofart-frame.jpeg ",
      "https://irisphoto.art/web/image/65508/19-98-31.jpg",
      "https://tds-images.thedailystar.net/sites/default/files/styles/amp_metadata_content_image_min_696px_wide/public/images/2022/10/14/ai_art_generator.png?itok=kgyM3PUE",
      "https://cdn.pixabay.com/photo/2017/08/30/12/45/girl-2696947_1280.jpg",
      "https://irisphoto.art/web/image/65508/19-98-31.jpg",
      "https://media.cdnws.com/_i/119489/433/3867/37/jm-arthot-newlessables-044-liberte-time-workofart-frame.jpeg ",
      "https://tds-images.thedailystar.net/sites/default/files/styles/amp_metadata_content_image_min_696px_wide/public/images/2022/10/14/ai_art_generator.png?itok=kgyM3PUE",
      "https://irisphoto.art/web/image/65508/19-98-31.jpg",
      "https://cdn.pixabay.com/photo/2017/08/30/12/45/girl-2696947_1280.jpg",
      "https://tds-images.thedailystar.net/sites/default/files/styles/amp_metadata_content_image_min_696px_wide/public/images/2022/10/14/ai_art_generator.png?itok=kgyM3PUE",
      "https://irisphoto.art/web/image/65508/19-98-31.jpg",
      "https://cdn.pixabay.com/photo/2017/08/30/12/45/girl-2696947_1280.jpg",
      "https://tds-images.thedailystar.net/sites/default/files/styles/amp_metadata_content_image_min_696px_wide/public/images/2022/10/14/ai_art_generator.png?itok=kgyM3PUE",
      "https://irisphoto.art/web/image/65508/19-98-31.jpg",
      "https://tds-images.thedailystar.net/sites/default/files/styles/amp_metadata_content_image_min_696px_wide/public/images/2022/10/14/ai_art_generator.png?itok=kgyM3PUE",
      "https://media.cdnws.com/_i/119489/433/3867/37/jm-arthot-newlessables-044-liberte-time-workofart-frame.jpeg ",
      "https://cdn.pixabay.com/photo/2017/08/30/12/45/girl-2696947_1280.jpg",
      "https://tds-images.thedailystar.net/sites/default/files/styles/amp_metadata_content_image_min_696px_wide/public/images/2022/10/14/ai_art_generator.png?itok=kgyM3PUE",
      "https://cdn.pixabay.com/photo/2017/08/30/12/45/girl-2696947_1280.jpg",
      "https://tds-images.thedailystar.net/sites/default/files/styles/amp_metadata_content_image_min_696px_wide/public/images/2022/10/14/ai_art_generator.png?itok=kgyM3PUE",
      "https://irisphoto.art/web/image/65508/19-98-31.jpg",
      "https://media.cdnws.com/_i/119489/433/3867/37/jm-arthot-newlessables-044-liberte-time-workofart-frame.jpeg ",
      "https://irisphoto.art/web/image/65508/19-98-31.jpg",
      "https://tds-images.thedailystar.net/sites/default/files/styles/amp_metadata_content_image_min_696px_wide/public/images/2022/10/14/ai_art_generator.png?itok=kgyM3PUE",
      "https://cdn.pixabay.com/photo/2017/08/30/12/45/girl-2696947_1280.jpg",
      "https://irisphoto.art/web/image/65508/19-98-31.jpg",
      "https://media.cdnws.com/_i/119489/433/3867/37/jm-arthot-newlessables-044-liberte-time-workofart-frame.jpeg ",
      "https://tds-images.thedailystar.net/sites/default/files/styles/amp_metadata_content_image_min_696px_wide/public/images/2022/10/14/ai_art_generator.png?itok=kgyM3PUE",
      "https://irisphoto.art/web/image/65508/19-98-31.jpg",
      "https://cdn.pixabay.com/photo/2017/08/30/12/45/girl-2696947_1280.jpg",
      "https://tds-images.thedailystar.net/sites/default/files/styles/amp_metadata_content_image_min_696px_wide/public/images/2022/10/14/ai_art_generator.png?itok=kgyM3PUE",
      "https://irisphoto.art/web/image/65508/19-98-31.jpg",
      "https://cdn.pixabay.com/photo/2017/08/30/12/45/girl-2696947_1280.jpg",
      "https://tds-images.thedailystar.net/sites/default/files/styles/amp_metadata_content_image_min_696px_wide/public/images/2022/10/14/ai_art_generator.png?itok=kgyM3PUE"
    ]
    setPictures(seedData);
  },[]);
  
  return (
    <div>
      <div className='flex justify-center'>
        <label className='text-6xl font-extrabold mt-7'><span className='leon-title'>Leon</span><span className='art-title'>'Art</span></label>
        <a href={redirectUrl}><button className='absolute mt-9 right-7 redirection-button text-xl font-extrabold'>{redirectText}</button></a>
      </div>
      <div className='flex justify-evenly mt-10'>
        <label className='text-4xl font-bold'>Artistes</label>
        <label className='text-4xl font-bold'>Cat√©gories</label>
        <label className='text-4xl font-bold'>Support</label>
        <label className='text-4xl font-bold'>Couleurs</label>
      </div>
      <div className='flex justify-center pb-10'>
        <div className='grid grid-cols-4 gap-8 place-items-center justify-center mt-8 mx-4'>
          {pictures.map((url, index) => (
            <div
              key={index}
              className="w-55 h-55 rounded-xl overflow-hidden bg-slate-300"
            >
              <img src={url} alt="" className="w-full h-full object-cover" />
            </div>
          ))}
        </div>
      </div>
    </div>
  );
}
  
export default Gallery;