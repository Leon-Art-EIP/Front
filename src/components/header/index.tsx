"use client";

import { ITab } from "../../../src/interfaces";
import Navbar from "../navbar/Navbar";
import Link from "../link/Link";
import { useEffect, useState } from "react";
import VerticalNavbar from "../navbar/VerticalNavbar";
import { myFetch } from "../../tools/myFetch";
import { IArtist } from "../../interfaces/home/artist";
import { imageApi } from "../../tools/variables";

interface IHeaderProps {
  tabs: ITab[];
  userId: string;
  pathname: string;
}

export default function Header(props: IHeaderProps): JSX.Element {
  const [width, setWindowWidth] = useState(0);
  const [profilePicture, setProfilePicture] = useState<string>("");

  const updateDimensions = () => {
    const width = window.innerWidth;
    setWindowWidth(width);
  };

  useEffect(() => {
    const fetchData = async () => {
      const response = await myFetch({ route: `/api/user/profile/${props.userId}`, method: "GET" });
      const artist = (await response.json()) as IArtist;
      setProfilePicture(artist.profilePicture);
    };

    fetchData();
    updateDimensions();
    window.addEventListener("resize", updateDimensions);
    return () => window.removeEventListener("resize", updateDimensions);
  }, [props.userId]);

  if (width < 500) {
    return <VerticalNavbar tabs={props.tabs} selectedTabHref={props.pathname} link={Link} userId={props.userId} />;
  }

  return (
    <Navbar
      tabs={props.tabs}
      selectedTabHref={props.pathname}
      link={Link}
      userId={props.userId}
      image={`${imageApi}/${profilePicture}`}
    />
  );
}
