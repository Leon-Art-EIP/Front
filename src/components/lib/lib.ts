export { default as NotificationToast } from './NotificationToast/NotificationToast';
export { default as Modal } from './Modal/Modal';
export { default as Card } from './Card/Card';
export { default as Popover } from './Popover/Popover';
export { default as Button } from './Buttons/Button';
export { default as ButtonGroup } from './Buttons/ButtonGroupToggle';
export { default as ButtonDropdown } from './Buttons/ButtonDopdown';
export { default as Badge } from './Badge/Badge';
export { default as Input } from './Input/Input';