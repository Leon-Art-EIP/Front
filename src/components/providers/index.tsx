"use client";

import { RecoilRoot, useRecoilValue } from "recoil";
import Session, { ISessionProps } from "../session";

export default function Providers(props: ISessionProps): JSX.Element {
  return (
    <RecoilRoot>
      <Session tabs={props.tabs}>{props.children}</Session>
    </RecoilRoot>
  );
}
