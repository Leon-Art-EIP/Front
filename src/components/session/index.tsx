"use client";

import { usePathname, useRouter, useSearchParams } from "next/navigation";
import { useEffect, useState } from "react";
import useFcmToken from "../../hooks/useFcmToken";
import { ITab } from "../../interfaces";
import { IConnectedUser } from "../../interfaces/user/user";
import { myFetch } from "../../tools/myFetch";
import Header from "../header";
import LoadingPage from "../loading/LoadingPage";

export interface ISessionProps {
  tabs: ITab[];
  children: React.ReactNode;
}

export default function Session(props: ISessionProps): JSX.Element | null {
  const { fcmToken, notificationPermissionStatus } = useFcmToken();
  const [user, setUser] = useState<IConnectedUser | null | undefined>(undefined);
  const pathname = usePathname();
  const router = useRouter();
  const searchParams = useSearchParams();
  const correspondingTab = props.tabs.find((tab) => {
    if (tab.href === "/") return tab.href === pathname;
    return pathname.includes(tab.href);
  });

  useEffect(() => {
    const token = searchParams.get("token");
    const username = searchParams.get("username");
    const email = searchParams.get("email");
    const id = searchParams.get("id");

    if (token && username && email && id) {
      const loggedInUser: IConnectedUser = {
        token,
        user: {
          id: id,
          username: username,
          email: email,
          is_artist: false,
          availability: "unavailable",
          subscription: "",
          collections: [],
        },
      };
      localStorage.setItem("user", JSON.stringify(loggedInUser));
      setUser(loggedInUser);

      if (fcmToken) {
        updateFcmTokenForUser(fcmToken);
      }
      // Redirect to the home page because this statement means that the user is logged in via google aouth2 redirection
      router.replace("/");
    } else {
      console.log("No token found in search params");
      const local = localStorage.getItem("user");
      if (local) {
        const loggedInUser = JSON.parse(local);
        setUser(loggedInUser);

        if (loggedInUser && loggedInUser.token && fcmToken) {
          updateFcmTokenForUser(fcmToken);
        }
      } else {
        setUser(null);
      }
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [pathname, searchParams, fcmToken]);

  useEffect(() => {
    if (correspondingTab?.loggedIn !== false && (user === null || (user && !user.token))) {
      router.replace("/login");
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [user, router]);

  if ((user && user.token) || ((!user || !user.token) && correspondingTab?.loggedIn === false)) {
    return (
      <>
        {correspondingTab?.header && user && <Header tabs={props.tabs} userId={user.user.id} pathname={pathname} />}
        {props.children}
      </>
    );
  }

  async function updateFcmTokenForUser(fcmToken: string) {
    const res = await myFetch({
      route: `/api/notifications/update-fcm-token`,
      method: "PUT",
      body: JSON.stringify({ fcmToken }),
    });
    if (!res.ok) {
      console.error("FCM token not updated");
    }
  }

  return <LoadingPage />;
}
