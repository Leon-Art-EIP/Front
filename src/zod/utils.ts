export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
