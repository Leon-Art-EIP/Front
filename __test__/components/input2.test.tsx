import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import { Input } from "../../src/components/lib";

test("Input snapshot - Basic usage", () => {
  testSnapshot(<Input type="text" value="Test value" placeHolder="Enter text" label="Text Input" />);
});

test("Input snapshot - Required input", () => {
  testSnapshot(<Input type="email" required label="Email" />);
});

test("Input snapshot - Disabled input", () => {
  testSnapshot(<Input type="text" value="Disabled" disabled label="Disabled Input" />);
});

test("Input snapshot - Without label", () => {
  testSnapshot(<Input type="text" value="No label" placeHolder="Placeholder only" />);
});

test("Input snapshot - Custom className", () => {
  testSnapshot(<Input type="number" value={123} className="custom-class" label="Number Input" />);
});
