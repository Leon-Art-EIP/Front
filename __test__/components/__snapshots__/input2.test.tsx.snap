// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input snapshot - Basic usage 1`] = `
<label
  className="undefined flex flex-col gap-2 text-tertiary font-regular text-md"
>
  <span>
    Text Input
     
  </span>
  <input
    className="text-md rounded-xl p-3 bg-gray-700 border-gray-600 placeholder-[#8F8F8F] text-white disabled:cursor-not-allowed disabled:text-gray"
    placeholder="Enter text"
    type="text"
    value="Test value"
  />
</label>
`;

exports[`Input snapshot - Custom className 1`] = `
<label
  className="custom-class flex flex-col gap-2 text-tertiary font-regular text-md"
>
  <span>
    Number Input
     
  </span>
  <input
    className="text-md rounded-xl p-3 bg-gray-700 border-gray-600 placeholder-[#8F8F8F] text-white disabled:cursor-not-allowed disabled:text-gray custom-class"
    type="number"
    value={123}
  />
</label>
`;

exports[`Input snapshot - Disabled input 1`] = `
<label
  className="undefined flex flex-col gap-2 text-tertiary font-regular text-md"
>
  <span>
    Disabled Input
     
  </span>
  <input
    className="text-md rounded-xl p-3 bg-gray-700 border-gray-600 placeholder-[#8F8F8F] text-white disabled:cursor-not-allowed disabled:text-gray"
    disabled={true}
    type="text"
    value="Disabled"
  />
</label>
`;

exports[`Input snapshot - Required input 1`] = `
<label
  className="undefined flex flex-col gap-2 text-tertiary font-regular text-md"
>
  <span>
    Email
     
    <span
      className="text-primary"
    >
      *
    </span>
  </span>
  <input
    className="text-md rounded-xl p-3 bg-gray-700 border-gray-600 placeholder-[#8F8F8F] text-white disabled:cursor-not-allowed disabled:text-gray"
    required={true}
    type="email"
  />
</label>
`;

exports[`Input snapshot - Without label 1`] = `
<label
  className="undefined flex flex-col gap-2 text-tertiary font-regular text-md"
>
  <span>
     
  </span>
  <input
    className="text-md rounded-xl p-3 bg-gray-700 border-gray-600 placeholder-[#8F8F8F] text-white disabled:cursor-not-allowed disabled:text-gray"
    placeholder="Placeholder only"
    type="text"
    value="No label"
  />
</label>
`;
